PAY-28795 - Membuat helper class untuk convert data type to / from Hex

ap-asep-agus-heri opened the pull request:
- Membuat Helper untuk encode ke hex dari byte array
- membuat converter string ke byte array untuk masing masing data type
- menambahkan pom xml
- membuat enum data type untuk mengetahui data type dari masing masing tag
- membuat enum tag indicator untuk mengetahui dan store tag dan subtag dari masing masing tag
- membuat unit test sebagai contoh implementasi penggunaan helper

ap-gracia-stefani committed:
modify pom

ap-gracia-stefani committed:
modify pom

ap-gracia-stefani committed:
testing refactoring class to mpm folder

ap-gracia-stefani committed:
skeleton

ap-gracia-stefani committed:
adjust structure

ap-asep-agus-heri committed:
PAY-28795 - Membuat helper class untuk convert data type to / from Hex

- Membuat Helper untuk encode ke hex dari byte array
- membuat converter string ke byte array untuk masing masing data type
- menambahkan pom xml
- membuat enum data type untuk mengetahui data type dari masing masing tag
- membuat enum tag indicator untuk mengetahui dan store tag dan subtag dari masing masing tag
- membuat unit test sebagai contoh implementasi penggunaan helper

ap-asep-agus-heri committed:
PAY-28795 - Membuat helper class untuk convert data type to / from Hex

- Membuat Helper untuk encode ke hex dari byte array
- membuat converter string ke byte array untuk masing masing data type
- menambahkan pom xml
- membuat enum data type untuk mengetahui data type dari masing masing tag
- membuat enum tag indicator untuk mengetahui dan store tag dan subtag dari masing masing tag
- membuat unit test sebagai contoh implementasi penggunaan helper

ap-asep-agus-heri committed:
PAY-28795 - Membuat helper class untuk convert data type to / from Hex

- Membuat Helper untuk encode ke hex dari byte array
- membuat converter string ke byte array untuk masing masing data type
- menambahkan pom xml
- membuat enum data type untuk mengetahui data type dari masing masing tag
- membuat enum tag indicator untuk mengetahui dan store tag dan subtag dari masing masing tag
- membuat unit test sebagai contoh implementasi penggunaan helper

ap-asep-agus-heri committed:
PAY-28795 - Membuat helper class untuk convert data type to / from Hex

- Membuat Helper untuk encode ke hex dari byte array
- membuat converter string ke byte array untuk masing masing data type
- menambahkan pom xml
- membuat enum data type untuk mengetahui data type dari masing masing tag
- membuat enum tag indicator untuk mengetahui dan store tag dan subtag dari masing masing tag
- membuat unit test sebagai contoh implementasi penggunaan helper

ap-asep-agus-heri committed:
PAY-28795 - Membuat helper class untuk convert data type to / from Hex

- Membuat Helper untuk encode ke hex dari byte array
- membuat converter string ke byte array untuk masing masing data type
- menambahkan pom xml
- membuat enum data type untuk mengetahui data type dari masing masing tag
- membuat enum tag indicator untuk mengetahui dan store tag dan subtag dari masing masing tag
- membuat unit test sebagai contoh implementasi penggunaan helper

ap-asep-agus-heri committed:
PAY-28795 - Membuat helper class untuk convert data type to / from Hex

- Membuat Helper untuk encode ke hex dari byte array
- membuat converter string ke byte array untuk masing masing data type
- menambahkan pom xml
- membuat enum data type untuk mengetahui data type dari masing masing tag
- membuat enum tag indicator untuk mengetahui dan store tag dan subtag dari masing masing tag
- membuat unit test sebagai contoh implementasi penggunaan helper

ap-asep-agus-heri committed:
PAY-28795 - Membuat helper class untuk convert data type to / from Hex

- Membuat Helper untuk encode ke hex dari byte array
- membuat converter string ke byte array untuk masing masing data type
- menambahkan pom xml
- membuat enum data type untuk mengetahui data type dari masing masing tag
- membuat enum tag indicator untuk mengetahui dan store tag dan subtag dari masing masing tag
- membuat unit test sebagai contoh implementasi penggunaan helper

ap-r-p-b-prameshwara commented on src/main/java/com/astrapay/qris/QrisHexConverter.java:
```diff
@@ -0,0 +1,102 @@
+    public String encode(byte[] value, byte[] tags) throws IOException {
+        var value_byteArray = new ByteArrayOutputStream();
+        //TAG INDICATOR
+        value_byteArray.write(tags[0]);
```
magic number at lines 37, 39, 40, 64, 66, 67

ap-asep-agus-heri replied:
done

ap-asep-agus-heri committed:
PAY-28795 - Membuat helper class untuk convert data type to / from Hex

- Membuat Helper untuk encode ke hex dari byte array
- membuat converter string ke byte array untuk masing masing data type
- menambahkan pom xml
- membuat enum data type untuk mengetahui data type dari masing masing tag
- membuat enum tag indicator untuk mengetahui dan store tag dan subtag dari masing masing tag
- membuat unit test sebagai contoh implementasi penggunaan helper
- uncomment

ap-asep-agus-heri committed:
PAY-28795 - Membuat helper class untuk convert data type to / from Hex

- Membuat Helper untuk encode ke hex dari byte array
- membuat converter string ke byte array untuk masing masing data type
- menambahkan pom xml
- membuat enum data type untuk mengetahui data type dari masing masing tag
- membuat enum tag indicator untuk mengetahui dan store tag dan subtag dari masing masing tag
- membuat unit test sebagai contoh implementasi penggunaan helper
- uncomment
- resolving magic number
